src/main/java/com/rackspace/jenkins_nodepool/NodePool.java

```diff
@@ -387,48 +387,25 @@ void provisionNode(Label label, Task task) throws Exception {
 
         // *** Poll request status and wait for fulfillment
         //TODO: store timeout in config
-        //TODO: Curator async stuff to avoid polling
-        final Integer timeout = 1200 * 1000; //timeout in milliseconds
-        final Long startTime = System.currentTimeMillis();
-        RequestState requestState = RequestState.requested;
+        final Integer timeout = 1200; //timeout in seconds
         List<NodePoolNode> allocatedNodes = null;
-        while (System.currentTimeMillis() < startTime + timeout) {
 
+        // Let's create a node pool watcher and wait until the request is in the desired state (or until we're timed out
+        final NodePoolRequestStateWatcher watcher = new NodePoolRequestStateWatcher(
+                conn.getZookeeperClient().getZooKeeper(), request.getPath(), RequestState.fulfilled);
```

Comment:
If you Passed in the NodeRequest object, you could use NodeRequest.updateFromZK instead of adding a new method for that.